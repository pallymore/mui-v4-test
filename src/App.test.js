import React from 'react';
import styled from 'styled-components';
import renderer from 'react-test-renderer';

import { ButtonBase } from '@material-ui/core';

const StyledButton = styled((props) => (
  <ButtonBase {...props} />
))`
  && {
    background-color: red;
  }
`;

const StyledNav = styled(({ active, component, ...rest }) => (
  <StyledButton component={component} {...rest} />
))``;

StyledNav.defaultProps = {
  component: 'a',
};

it('renders active without crashing', () => {
  const tree = renderer.create(<StyledNav active>NavItem</StyledNav>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders disabled without crashing', () => {
  const tree = renderer.create(<StyledNav disabled>NavItem</StyledNav>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders as button without crashing', () => {
  const tree = renderer.create(<StyledNav component="button">NavItem</StyledNav>).toJSON();
  expect(tree).toMatchSnapshot();
});

