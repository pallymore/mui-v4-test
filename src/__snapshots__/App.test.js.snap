// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders active without crashing 1`] = `
<a
  className="MuiButtonBase-root sc-bwzfXH uTxCW sc-bdVaJa bIidUE"
  onBlur={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="button"
  tabIndex={0}
>
  NavItem
</a>
`;

exports[`renders as button without crashing 1`] = `
<button
  className="MuiButtonBase-root sc-bwzfXH uTxCW sc-bdVaJa bIidUE"
  onBlur={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  NavItem
</button>
`;
